contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -158,56 +154,86 @@ contract VaultHub is PausableUntilWithRoles {
+    /// @return connection parameters struct for the given vault
+    /// @dev it returns empty struct if the vault is not connected to the hub
+    /// @dev it may return connection even if it's pending to be disconnected
+    function vaultConnection(address _vault) external view returns (VaultConnection memory) {
+        return _storage().connections[_vault];
+    }
+
+    /// @return the accounting record for the given vault
+    /// @dev it returns empty struct if the vault is not connected to the hub
+    function vaultRecord(address _vault) external view returns (VaultRecord memory) {
+        return _storage().records[_vault];
     }
 
     /// @return total value of the vault (as of the latest report received)
+    /// @dev returns 0 if the vault is not connected
     function totalValue(address _vault) external view returns (uint256) {
-        VaultSocket storage socket = _connectedSocket(_vault);
-        return _totalValue(socket);
+        return _totalValue(_storage().records[_vault]);
+    }
+
+    /// @return liability shares of the vault
+    /// @dev returns 0 if the vault is not connected
+    function liabilityShares(address _vault) external view returns (uint256) {
+        return _storage().records[_vault].liabilityShares;
+    }
+
+    /// @return locked amount of ether for the vault
+    /// @dev returns 0 if the vault is not connected
+    function locked(address _vault) external view returns (uint256) {
+        return _storage().records[_vault].locked;
     }
 
     /// @return amount of ether that is part of the vault's total value and is not locked as a collateral
+    /// @dev returns 0 if the vault is not connected
     function unlocked(address _vault) external view returns (uint256) {
-        VaultSocket storage socket = _connectedSocket(_vault);
-        return _unlocked(socket);
+        return _unlocked(_storage().records[_vault]);
+    }
+
+    /// @return the latest report for the vault
+    /// @dev returns empty struct if the vault is not connected
+    function latestReport(address _vault) external view returns (Report memory) {
+        return _storage().records[_vault].report;
     }
 
     /// @return true if the report for the vault is fresh, false otherwise
+    /// @dev returns false if the vault is not connected
     function isReportFresh(address _vault) external view returns (bool) {
-        VaultSocket storage socket = _connectedSocket(_vault);
-        return _isReportFresh(socket);
+        return _isReportFresh(_storage().records[_vault]);
     }
 
     /// @notice checks if the vault is healthy by comparing its total value after applying rebalance threshold
     ///         against current liability shares
     /// @param _vault vault address
     /// @return true if vault is healthy, false otherwise
+    /// @dev reverts if the vault is not connected
     function isVaultHealthyAsOfLatestReport(address _vault) external view returns (bool) {
```

Comment:
🤔  `isVaultHealthyAsOfLatestReport` is wrong, it can change to healthy if the inOutDelta increases enough